### PR TITLE
Add TicketTailor integration + EventList block

### PR DESCRIPTION
**Pull request by janbaykara**
Allows importing data from TicketTailor.

## Description
- [x] Adds `TicketTailorSource` and connection boilerplate
- [x] Adds new flags, tests, code and UI guard-rails for read-only data sources
- [x] Adds an `EventList` block in Puck to display event data. You can select a data source and filter out certain events.
- [ ] One outstanding task is to add import webhook support. We need webhook logic for triggering imports when these webhooks come in, and we need UI to display the webhook URL to the user so they can enable it.

## Motivation and Context

TicketTailor is used by the Climate Coalition to organise RSVPs for their training schedule and, alongside EventBrite, is one of the few big ticketing/events platforms.

Events are a common type of geographically mappable entities, given they generally a location and a time. But even if they are online, a public hub should route people to relevant events.

This PR makes it possible to advertise events from these third parties in curated, dynamic maps and timelines.

It also sets the scene for other read-only data sources like EventBrite, hosted CSVs, JSON APIs, SQL databases and much else.

## How Can It Be Tested?

You can run the tests locally by adding `TEST_TICKET_TAILOR_API_KEY` to your .env.

Bitwarden login for https://tickettailor.com/ is jan@commonknowledge.coop — you can find the API key in `Box Office Settings` > `API`

1. Connect TicketTailor as a CRM — you should have no options beyond the name
2. In the inspector, there should be no update options
3. Import data
4. Create a map and display some events there

## How Will This Be Deployed?

N/A

## Screenshots (if appropriate):

Screenshot 2024-06-03 at 01 08 00

## Types of changes
- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx [hunk 1]
```diff
@@ -94,23 +95,7 @@ export default function Page({
 
   const form = useForm<FormInputs>({
     defaultValues: {
-      name: '',
       geographyColumnType: GeographyTypes.Postcode,
```
Comment (joaquimds) [inline nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx:98]: suggestion: we should keep these default variables to squash console error logs, but restrict them based on the externalDataSourceType.

Comment (janbaykara) [reply nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx:99]: OK, I can revert these changes.

Comment (janbaykara) [reply nextjs/src/app/(app)/data-sources/create/connect/[externalDataSourceType]/page.tsx:126]: Fixed. Noticed a separate bug where postcode field doesn't get prefilled anymore by guesser.

**Comment by janbaykara** [comment]
I feel the URL and instruction should appear after connection. There are two instructions for sources that don't have automated webhooks:

1. The import URL and instruction
2. For sources that allow updates, the import URL and instruction

Could also be a follow-on PR.

**Pull request merged**